apollo-client/src/main/java/com/ctrip/framework/apollo/internals/PureApolloConfig.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2021 Apollo Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package com.ctrip.framework.apollo.internals;
+
+import com.ctrip.framework.apollo.util.ApolloHashMapInitialUtil;
+import java.util.HashSet;
+import java.util.Set;
+import org.springframework.util.CollectionUtils;
+
+/**
+ * @author vdisk <vdisk@foxmail.com>
+ */
+public class PureApolloConfig extends AbstractRepositoryConfig implements RepositoryChangeListener {
+
+  /**
+   * Constructor.
+   *
+   * @param namespace        the namespace of this config instance
+   * @param configRepository the config repository for this config instance
+   */
+  public PureApolloConfig(String namespace,
+      ConfigRepository configRepository) {
+    super(namespace, configRepository);
+  }
+
+  @Override
+  public String getProperty(String key, String defaultValue) {
```

Comment:
It looks like the only difference between PureApolloConfig and DefaultConfig is the getProperty method, then is it simpler to let PureApolloConfig extends DefaultConfig and override the getProperty method?

Reply:
resolved